crates/bitwarden-crypto/src/keys/signed_public_key.rs

```diff
@@ -0,0 +1,181 @@
+//! A public encryption key alone is not authenticated. It needs to be tied to a cryptographic
+//! identity, which is provided by a signature keypair. This is done by signing the public key, and
+//! requiring consumers to verify the public key before consumption by using unwrap_and_verify.
+
+use std::str::FromStr;
+
+use base64::{engine::general_purpose::STANDARD, Engine};
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+use super::AsymmetricPublicCryptoKey;
+use crate::{
+    cose::CoseSerializable, error::EncodingError, util::FromStrVisitor, CryptoError,
+    PublicKeyEncryptionAlgorithm, RawPublicKey, SignedObject, SigningKey, SigningNamespace,
+    VerifyingKey,
+};
+
+#[cfg(feature = "wasm")]
+#[wasm_bindgen::prelude::wasm_bindgen(typescript_custom_section)]
+const TS_CUSTOM_TYPES: &'static str = r#"
+export type SignedPublicKey = string;
+"#;
+
+/// `PublicKeyFormat` defines the format of the public key in a `SignedAsymmetricPublicKeyMessage`.
+/// Currently, only ASN.1 Subject Public Key Info (SPKI) is used, but CoseKey may become another
+/// option in the future.
+#[derive(Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+enum PublicKeyFormat {
+    Spki = 0,
+}
+
+/// `SignedAsymmetricPublicKeyMessage` is a message that once signed, makes a claim towards owning a
+/// public encryption key.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignedPublicKeyMessage {
+    /// The algorithm/crypto system used with this public key.
+    algorithm: PublicKeyEncryptionAlgorithm,
+    /// The format of the public key.
+    content_format: PublicKeyFormat,
+    /// The public key, serialized and formatted in the content format specified in
+    /// `content_format`.
+    ///
+    /// Note: `ByteBuf` is used here to ensure efficient serialization. Using `Vec<u8>` would
```

Comment:
We should use square brackets instead so we get links in the code docs
```suggestion
    /// Note: [ByteBuf] is used here to ensure efficient serialization. Using [Vec<u8>] would
```

Reply:
Apparently you have to use the back quotes inside the brackets  as well if there are angle brackets in the type, my bad ```[`Vec<u8>`]```

Reply:
No worries. Fixed!